src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementBackedDocumentsService.java

```diff
@@ -94,28 +101,30 @@ private Claim getClaimByExternalId(String externalId, String authorisation) {
         return claimService.getClaimByExternalId(externalId, authorisation);
     }
 
-    private DocumentDetails uploadToDocumentManagement(
+    @Override
+    public byte[] uploadToDocumentManagement(
         byte[] documentBytes,
         String authorisation,
-        String baseFileName) {
-        PDF document = new PDF(baseFileName, documentBytes);
+        String baseFileName,
+        ClaimDocumentType claimDocumentType,
+        Claim claim) {
+        PDF document = new PDF(baseFileName, documentBytes, claimDocumentType);
         URI documentSelfPath = documentManagementService.uploadDocument(authorisation, document);
-        return new DocumentDetails() {
-            @Override
-            public URI getDocumentSelfPath() {
-                return documentSelfPath;
-            }
-
-            @Override
-            public PDF getDocument() {
-                return document;
-            }
-        };
+        claimService.linkClaimToDocument(authorisation,
+            claim.getId(),
+            getClaimDocumentStore(claim.getExternalId(), document, documentSelfPath, authorisation));
+        return documentBytes;
     }
 
-    private interface DocumentDetails {
-        URI getDocumentSelfPath();
-
-        PDF getDocument();
+    private ClaimDocumentStore getClaimDocumentStore(String externalId, PDF document, URI uri, String authorisation) {
+        Claim claim = claimService.getClaimByExternalId(externalId, authorisation);
+        ClaimDocumentStore claimDocumentStore = claim.getClaimDocumentStore().orElse(new ClaimDocumentStore());
+        claimDocumentStore.addClaimDocument(new ClaimDocument(uri,
+            document.getFilename(),
+            document.getClaimDocumentType(),
+            LocalDateTimeFactory.nowInLocalZone(),
+            LocalDateTimeFactory.nowInLocalZone(),
+            null));
```

Comment:
can this null be specific as What it meant, like extract as constant and give meaningful name and initialised to null

Reply:
Can use builder instead of passing null.